DZ_6/app_01.py

```diff
@@ -0,0 +1,288 @@
+import datetime
+from typing import List
+
+
+import databases
+import pandas as pd
+import sqlalchemy
+from fastapi import FastAPI, Request, HTTPException
+from fastapi.responses import HTMLResponse
+from sqlalchemy import create_engine, Column, Integer, String, DateTime, ForeignKey
+from sqlalchemy.ext.declarative import declarative_base
+from pydantic import BaseModel, Field
+from fastapi.templating import Jinja2Templates
+from sqlalchemy import DATE, Float
+
+
+
+DATABASE_URL = "sqlite:///shop.db"
+
+database = databases.Database(DATABASE_URL)
+metadata = sqlalchemy.MetaData()
+
+goods = sqlalchemy.Table(
+    "goods",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("goods_name", sqlalchemy.String(40)),
+    sqlalchemy.Column("goods_text", sqlalchemy.String(120)),
+    sqlalchemy.Column("price", sqlalchemy.Float),
+)
+
+orders = sqlalchemy.Table(
+    "orders",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("user_id", sqlalchemy.Integer, sqlalchemy.ForeignKey("users.id")),
+    sqlalchemy.Column("goods_id", sqlalchemy.Integer, sqlalchemy.ForeignKey("goods.id")),
+    sqlalchemy.Column("order_data", sqlalchemy.DATE),
+    sqlalchemy.Column("status", sqlalchemy.String(40))
+)
+
+users = sqlalchemy.Table(
+    "users",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String(40)),
+    sqlalchemy.Column("surname", sqlalchemy.String(40)),
+    sqlalchemy.Column("email", sqlalchemy.String(120)),
+    sqlalchemy.Column("password", sqlalchemy.Integer),
+
+)
+
+engine = sqlalchemy.create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
+metadata.create_all(engine)
+
+app = FastAPI()
+templates = Jinja2Templates(directory="./DZ_6/templates")
+
+
+class User(BaseModel):
+    id: int
+    name: str = Field(max_length=40)
+    surname: str = Field(max_length=40)
+    email: str = Field(max_length=120)
+    password: str = Field(max_length=5)
+
+class UserIn(BaseModel):
+    name: str = Field(max_length=40)
+    surname: str = Field(max_length=40)
+    email: str = Field(max_length=120)
+    password: str = Field(max_length=5)
```

Comment:
тут можно заметить, что классы практически идентичны, а значит, один может наследоваться от другого.
```suggestion
class UserIn(BaseModel):
    name: str = Field(max_length=40)
    surname: str = Field(max_length=40)
    email: str = Field(max_length=120)
    password: str = Field(max_length=5)
    
    
class User(UserIn):
    id: int
```